src/drivers/gps/gps.cpp

```diff
@@ -810,15 +800,39 @@ GPS::print_info()
 	usleep(100000);
 }
 
+void
+GPS::publish()
+{
+	if (_gps_num == 1) {
+		if (_report_gps_pos_pub[0] != nullptr) {
+			orb_publish(ORB_ID(vehicle_gps_position), _report_gps_pos_pub[0], &_report_gps_pos);
+
+		} else {
+			_report_gps_pos_pub[0] = orb_advertise_multi(ORB_ID(vehicle_gps_position), &_report_gps_pos, &_gps_orb_instance[0], ORB_PRIO_DEFAULT);
```

Comment:
@bkueng This is way too complex. It should be orb_publish_auto() instead, which will take care of the multi instance handling and noes not require the nullptr checking.